src/main/java/com/google/api/codegen/InterfaceConfig.java

```diff
@@ -76,18 +77,32 @@ public static InterfaceConfig createInterfaceConfig(
       methodConfigs = createMethodConfigs(methodConfigMap, interfaceConfigProto);
     }
 
+    SmokeTestConfig smokeTestConfig =
+        createSmokeTestConfig(diagCollector, iface, interfaceConfigProto);
+
```

Comment:
I take it back, 'null' is not the right signal. I think we should change the check below to check whether there are errors on diagCollector (and return 'null' if so).